packages/core/src/components/DataTable/types.tsx

```diff
@@ -0,0 +1,251 @@
+  sortByOverride?: string;
+  /** sortDirection value if override is enabled. */
+  sortDirectionOverride?: SortDirectionType;
+  /** sortCallback in override is enabled. */
+  sortCallback?: (sortBy: string, sortDirection: SortDirectionType) => void;
+  /** Label for the table header. */
+  tableHeaderLabel?: string;
+  /** Height of the table header. */
+  tableHeaderHeight?: HeightOptions;
+  /** Width of the entire table. */
+  width?: number;
+  /** If enabled, every other row will appear in grey. */
+  zebra?: boolean;
+}
+
+export interface GenericRow {
+  data: {
+    [key: string]: any;
+  };
+}
+
+/** The row used by React Virtualized. */
+export interface TableRow {
+  cellData?: void;
+  columnData?: void;
+  columnIndex: number;
+  dataKey: string;
+  isScrolling?: boolean;
+  parent?: React.ReactNode;
+  rowData: ExpandedRow;
+  rowIndex: number;
+}
+
+export type Status = string;
+
+type ParentMetadata = {
+  colspanKey?: string;
+  children?: ChildRow[];
+  status?: string;
+};
+
+type ChildMetadata = {
+  colspanKey?: string;
+  status?: Status;
+};
+
+type IndexedParentMetadata = ParentMetadata & {
+  originalIndex: number;
+  isChild: boolean;
+  children: IndexedChildRow[];
+};
+
+type IndexedChildMetadata = ChildMetadata & {
+  originalIndex: number;
+  children?: undefined;
+  isChild: boolean;
+};
+
+type ExpandedParentMetadata = IndexedParentMetadata & {
+  preExpandedIndex: number;
+  parentIndex?: undefined;
+  parentOriginalIndex?: undefined;
+};
+
+type ExpandedChildMetadata = IndexedChildMetadata & {
+  preExpandedIndex?: undefined;
+  parentIndex: number;
+  parentOriginalIndex: number;
+};
+
+interface Row {
+  data: {
+    [key: string]: any;
+  };
+}
+
+export interface ChildRow extends Row {
+  metadata?: ChildMetadata;
+}
+
+export interface ParentRow extends Row {
+  metadata?: ParentMetadata;
+}
+
+export interface IndexedParentRow extends Row {
+  metadata: IndexedParentMetadata;
+}
+export interface IndexedChildRow extends Row {
+  metadata: IndexedChildMetadata;
+}
+
+export interface ExpandedParentRow extends Row {
+  metadata: ExpandedParentMetadata;
+}
+export interface ExpandedChildRow extends Row {
+  metadata: ExpandedChildMetadata;
+}
+
+export type IndexedRow = IndexedChildRow | IndexedParentRow;
+export type ExpandedRow = ExpandedParentRow | ExpandedChildRow;
+
+export type RowStyles = {
+  [key: string]: string;
+};
+
+export type ColumnMetadata = {
+  [key: string]: {
+    [key: string]: number;
+  };
+};
+
+export type ColumnToLabel = {
+  [key: string]: string;
+};
+
+export type RendererProps = {
+  /** Row including row data and metadata. */
+  row: TableRow;
+  /** Key being rendered. */
+  key: string;
+  /** Whether or not edit mode is enabled. */
+  editMode: boolean;
+  /** Callback to trigger on cell edit. */
+  handleEdit: HandleEdit;
+  /** Whether or not zebra mode is enabled. */
+  zebra: boolean;
+  /** Theme from Lunar. */
+  theme: WithStylesProps['theme'];
```

Comment:
Is this necessary?

Reply:
I use it in `ColspanRenderer` and can imagine it being necessary for user-defined custom renderers in the future. It's sort of unfortunate, but I definitely want to give people access to getRowColor in their renderers, which means they'll need this prop.